shithead_game_server/src/lobby.rs

```diff
@@ -109,44 +122,192 @@ impl Deck {
         }
         Some(self.cards.drain(self.cards.len() - amount..))
     }
+
+    /// Takes all the cards from the deck.
+    pub fn take_all<'a>(&'a mut self) -> std::vec::Drain<'a, CardId> {
+        self.cards.drain(..)
+    }
+}
+
+/// A timer which waits for the next turn, and once the time is out, switches the turn.
+#[derive(Debug)]
+pub struct NextTurnTimer {
+    task: tokio::task::JoinHandle<()>,
+    notify: Arc<Notify>,
+}
+impl NextTurnTimer {
+    pub async fn stop(self) {
+        self.notify.notify_one();
+        self.task.await.unwrap();
```

Comment:
Does unwrapping an await cancels the await on the timer?

Reply:
No, awaiting just waits for it to finish, and unwrapping makes sure the task didn't panic. If the task panicked, which shouldn't happen, the unwrap will cause the current task to crash as well. This is useful so that if there is some logic bug in the code which made the timer task crash, we would like to know about it, and not let it go in  silence.